builder/cloudstack/config.go

```diff
@@ -40,6 +40,8 @@ type Config struct {
 	PublicIPAddress      string   `mapstructure:"public_ip_address"`
 	SecurityGroups       []string `mapstructure:"security_groups"`
 	ServiceOffering      string   `mapstructure:"service_offering"`
+	CreateNetworkACL     bool     `mapstructure:"create_network_acl"`
+	CreateFirewallRule   bool     `mapstructure:"create_firewall_rule"`
```

Comment:
What would be the reason (the use case) to prevent creating the firewall rules?

If there is a good reason to add something like this, then I think we should only create one option which, if not specified, doesn't change the existing functionality. The way they are created in this PR means that everyone needs to add these fields  to their config and set the value to `true` to not break stuff.

Reply:
There are many possible configurations and customisations for cloudstack, for my personal use case, I needed to disable that step to avoid messing up with existing rules. Let me have a look into this one to see if I can consolidate into 1 single flag.

Reply:
I cannot come up with a name to consolidate both, in fact, I think it can be useful for everyone to have that granularity to disable the creation of acl and/or firewall rules if needed.

I can set the default value to `true` in `NewConfig` function before mapping the config, so it doesn't modify the default behaviour.

```
        c := new(Config)

	// By default builder will try to create either network acls or firewall rules
	c.CreateNetworkACL = true
	c.CreateFirewallRule = true

	err := config.Decode(c, &config.DecodeOpts{
		Interpolate:        true,
		InterpolateContext: &c.ctx,
		InterpolateFilter: &interpolate.RenderFilter{
			Exclude: []string{
				"user_data",
			},
		},
	}, raws...)
	if err != nil {
		return nil, err
	}
```

Do you like this solution or do you want me to do something else?

Reply:
Maybe call it `PreventFirewallChanges` and then make sure the docs explain that this prevents any changes your network ACL or firewall (depending on the type of networking you are using).

And then this option could be used on line 119 of the networking file (step_configure_networking.go) like:

```go
if config.PreventFirewallChanges {
  ui.Message("Networking has been setup (without making firewall changes)!")
  return multistep.ActionContinue
}
```

Reply:
Great, I will work towards that idea 👍